<a href="https://colab.research.google.com/github/Malu247/programacion1-/blob/main/quiz26marzo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Quiz 26 marzo 2026

Maria Lucia Higuera Angarita 2260674

In [21]:
%%writefile pedirnotas.py
def pedir_nombre():
  return input("Ingrese nombre: ")

def pedir_notas():
  n1 = float(input("Nota 1: "))
  n2 = float(input("Nota 2: "))
  n3 = float(input("Nota 3: "))
  n4 = float(input("Nota 4: "))
  return n1, n2, n3, n4

Overwriting pedirnotas.py


In [22]:
import pedirnotas
nombre = pedirnotas.pedir_nombre()


Ingrese nombre: Maria


In [23]:
import importlib
import pedirnotas
importlib.reload(pedirnotas)
notas = pedirnotas.pedir_notas()

Nota 1: 2
Nota 2: 3
Nota 3: 3
Nota 4: 5


In [26]:
%%writefile promedio.py

import statistics
def calcular_promedio(n1,n2,n3,n4):
  return (n1+n2+n3+n4)/4

def mediana(n1,n2,n3,n4):
  return statistics.median([n1,n2,n3,n4])

def moda(n1,n2,n3,n4):
  return statistics.mode([n1,n2,n3,n4])

Overwriting promedio.py


In [27]:
import promedio
importlib.reload(promedio)
prom = promedio.calcular_promedio(notas[0], notas[1], notas[2], notas[3])
mediana = promedio.mediana(notas[0], notas[1], notas[2], notas[3])
moda = promedio.moda(notas[0], notas[1], notas[2], notas[3])

print("EStudiante: ", nombre)
print("Promedio: ", prom)
print("Mediana", mediana)
print("Moda", moda)

if prom >=3:
  print("Aprobado")

else:
  print("Reprobado")

EStudiante:  Maria
Promedio:  3.25
Mediana 3.0
Moda 3.0
Aprobado


* Media aritmética: En promedio el estudiante obtuvo 3.25
* Mediana: El valor que divide el 50% de las calificaciones es 3
* Moda: La calificación más repetida es 3
* El desempeño del estudiante es **básico** porque su promedio estuvo solo un poco por encima de la nota mínima aprobatoria.
